File: semestru 4/MotoProject/Networking/src/main/java/network/dto/DTOUtils.java
package network.dto;


import model.Player;
import model.Race;
import model.Team;
import model.User;

import java.util.Optional;

public class DTOUtils {
    public static User getFromDTO(UserDTO usdto){
        String username=usdto.getUsername();
        String pass=usdto.getPasswd();

        return new User(username,pass);

    }
    public static User[] getFromDTO(UserDTO[] users){
        User[] friends=new User[users.length];
        for(int i=0;i<users.length;i++){
            friends[i]=getFromDTO(users[i]);
        }
        return friends;
    }
    public static UserDTO getDTO(User user){
        String username=user.getUsername();
        String pass=user.getPassword();
        return new UserDTO(username, pass);
    }
    public static UserDTO[] getDTO(User[] users){
        UserDTO[] frDTO=new UserDTO[users.length];
        for(int i=0;i<users.length;i++)
            frDTO[i]=getDTO(users[i]);
        return frDTO;
    }


    public static PlayerDTO getDTO(Player player){
        String name=player.getName();
        String code=player.getCode();
        String team = player.getTeam().toString();
        return new PlayerDTO(name, code,team);
    }
    public static PlayerDTO[] getDTO(Player[] players) {
        PlayerDTO[] pDTO=new PlayerDTO[players.length];
        for(int i=0;i<players.length;i++)
            pDTO[i]=getDTO(players[i]);
        return pDTO;
    }
    public static Player getFromDTO(PlayerDTO playerDTO){
        String name=playerDTO.getName();
        String code=playerDTO.getCode();
        Integer team=Integer.parseInt(playerDTO.getTeam());
        return new Player(name, code, team);
    }
    public static Player[] getFromDTO(PlayerDTO[] players) {
        Player[] pDTO=new Player[players.length];
        for(int i=0;i<players.length;i++)
            pDTO[i]=getFromDTO(players[i]);
        return pDTO;
    }
    public static RaceDTO getDTO(Race race){
        RaceDTO raceDTO=new RaceDTO();
        raceDTO.setId(String.valueOf(race.getId()));
        raceDTO.setEngineType(String.valueOf(race.getEngineType()));
        raceDTO.setNoPlayers(String.valueOf(race.getNoPlayers()));
        return raceDTO;
    }
    public static RaceDTO[] getDTO(Race[] races) {
        RaceDTO[] rDTO=new RaceDTO[races.length];
        for(int i=0;i<races.length;i++)
            rDTO[i]=getDTO(races[i]);
        return rDTO;
    }

    public static Race getFromDTO(RaceDTO raceDTO){
        Integer engineType=Integer.parseInt(raceDTO.getEngineType());
        Integer noPlayers=Integer.parseInt(raceDTO.getNoPlayers());
        Integer id=Integer.parseInt(raceDTO.getId());
        Race race = new Race(engineType);
        race.setId(id);
        race.setNoPlayers(noPlayers);
        return race;

    }
    public static Race[] getFromDTO(RaceDTO[] races) {
        Race[] rDTO=new Race[races.length];
        for(int i =0; i <races.length;i++){
            rDTO[i]=getFromDTO(races[i]);
        }
        return rDTO;
    }



    public static Team getFromDTO(TeamDTO team) {
        Team t = new Team(team.getName());
        t.setId(Integer.valueOf(team.getId()));
        return t;
    }

    public static TeamDTO getDTO(Team team) {
        TeamDTO tDTO = new TeamDTO(team.getName());
        tDTO.setId(String.valueOf(team.getId()));
        return tDTO;
    }

}

File: semestru 4/MotoProjectCs/networking/dto/DTOUtils.cs
using moto.model;

namespace moto.networking.dto;

public class DTOUtils
{
    public static User GetFromDTO(UserDTO usdto)
    {
        string username = usdto.Username;
        string pass = usdto.Passwd;

        return new User(username, pass);
    }

    public static User[] GetFromDTO(UserDTO[] users)
    {
        User[] friends = new User[users.Length];
        for (int i = 0; i < users.Length; i++)
        {
            friends[i] = GetFromDTO(users[i]);
        }
        return friends;
    }

    public static UserDTO GetDTO(User user)
    {
        string username = user.Username;
        string pass = user.Password;
        return new UserDTO(username, pass);
    }

    public static UserDTO[] GetDTO(User[] users)
    {
        UserDTO[] frDTO = new UserDTO[users.Length];
        for (int i = 0; i < users.Length; i++)
        {
            frDTO[i] = GetDTO(users[i]);
        }
        return frDTO;
    }

    public static PlayerDTO GetDTO(Player player)
    {
        string name = player.Name;
        string code = player.Code;
        string team = player.Team.ToString();
        return new PlayerDTO(name, code, team);
    }

    public static PlayerDTO[] GetDTO(Player[] players)
    {
        PlayerDTO[] pDTO = new PlayerDTO[players.Length];
        for (int i = 0; i < players.Length; i++)
        {
            pDTO[i] = GetDTO(players[i]);
        }
        return pDTO;
    }

    public static Player GetFromDTO(PlayerDTO playerDTO)
    {
        string name = playerDTO.Name;
        string code = playerDTO.Code;
        int team = int.Parse(playerDTO.Team);
        return new Player(name, code, team);
    }

    public static Player[] GetFromDTO(PlayerDTO[] players)
    {
        Player[] pDTO = new Player[players.Length];
        for (int i = 0; i < players.Length; i++)
        {
            pDTO[i] = GetFromDTO(players[i]);
        }
        return pDTO;
    }

    public static RaceDTO GetDTO(Race race)
    {
        string Id = race.Id.ToString();
        string EngineType = race.EngineType.ToString();
        string NoPlayers = race.NoPlayers.ToString();
        RaceDTO raceDTO = new RaceDTO(NoPlayers, EngineType, Id);
        return raceDTO;
    }

    public static RaceDTO[] GetDTO(Race[] races)
    {
        RaceDTO[] rDTO = new RaceDTO[races.Length];
        for (int i = 0; i < races.Length; i++)
        {
            rDTO[i] = GetDTO(races[i]);
        }
        return rDTO;
    }

    public static Race GetFromDTO(RaceDTO raceDTO)
    {
        int engineType = int.Parse(raceDTO.EngineType);
        int noPlayers = int.Parse(raceDTO.NoPlayers);
        int id = int.Parse(raceDTO.Id);
        Race race = new Race(engineType)
        {
            Id = id,
            NoPlayers = noPlayers
        };
        return race;
    }

    public static Race[] GetFromDTO(RaceDTO[] races)
    {
        Race[] rDTO = new Race[races.Length];
        for (int i = 0; i < races.Length; i++)
        {
            rDTO[i] = GetFromDTO(races[i]);
        }
        return rDTO;
    }

    public static Team GetFromDTO(TeamDTO team)
    {
        Team t = new Team(team.Name)
        {
            Id = int.Parse(team.Id)
        };
        return t;
    }

    public static TeamDTO GetDTO(Team team)
    {
        TeamDTO tDTO = new TeamDTO(team.Name)
        {
            Id = team.Id.ToString()
        };
        return tDTO;
    }
}